enterprise/utils.py

```diff
@@ -1142,6 +1142,36 @@ def get_course_run_duration_info(course_run):
     return duration_info
 
 
+def get_duration_of_course_or_courserun(content_metadata_item):
+    """
+    Returns duration start, end dates given a piece of content_metadata item
+    If course item, extracts start, end dates of closest course run based on current timestamp
+    Returns:
+        duration: in days or 0
+        start: start field of closest course run item, or None
+        end: end field of closest course run item, or None
+    """
+    start = None
+    end = None
+    if content_metadata_item.get('content_type') == 'courserun':
+        start = content_metadata_item.get('start')
+        end = content_metadata_item.get('end')
+    elif content_metadata_item.get('content_type') == 'course':
+        course_runs = content_metadata_item.get('course_runs')
+        if course_runs:
```

Comment:
when would we see no course runs in the metadata?

Reply:
I actually don't know if this happens in real life, but it can happen and is just defensive programming. Also noted this pattern existed in some other code in integ channels and/or utils.

Reply:
also fwiw: this pattern of 'find a course run if it exists, and extract some field(s) out of it' also seems fairly repeated so perhaps there is an opportunity for us to try DRY too... just did not want to go overboard in this PR